src/lib.rs

```diff
@@ -165,15 +165,20 @@ impl UPTransportZenoh {
         } else {
             uri.authority_name.clone()
         };
-        // ue_id
-        let ue_id = if uri.has_wildcard_entity_type() {
-            // If entity type is wildcard, the whole entity should be wilrdcard
-            "*".to_string()
-        } else if uri.has_wildcard_entity_instance() {
-            // If entity instance is wildcard, we still need to check the entity type
-            format!("$*{:04X}", uri.ue_id & WILDCARD_ENTITY_TYPE)
+        // ue_type
+        let (ue_type, ue_instance) = if uri.has_wildcard_entity_type() {
+            // If entity type is wildcard, the instance should also be wilrdcard
+            ("*".to_string(), "*".to_string())
         } else {
-            format!("{:X}", uri.ue_id)
+            let ue_instance = if uri.has_wildcard_entity_instance() {
+                "*".to_string()
+            } else {
+                format!("{:X}", uri.ue_id >> 16)
```

Comment:
It seems like I can leverage https://github.com/eclipse-uprotocol/up-rust/blob/2cb2e112e6f497ccb4d472157becc572fd78ac1c/src/uri.rs#L425
But it will need https://github.com/eclipse-uprotocol/up-transport-zenoh-rust/pull/98 to be merged anyway.

Reply:
indeed